packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -315,37 +315,26 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
     );
   }
 
-  DataRow _getProgressIndicatorRowFor(int index) {
-    bool haveProgressIndicator = false;
-    final List<DataCell> cells = widget.columns.map<DataCell>((DataColumn column) {
-      if (!column.numeric) {
-        haveProgressIndicator = true;
-        return const DataCell(CircularProgressIndicator());
+  bool _haveProgressIndicator(int firstRowIndex, int rowsPerPage) {
```

Comment:
This can probably merged into get merge into _getRows? maybe make _getRows a more generic function like _updateRowsInCurrentPage to avoid widget.source.getRow called  twice for every index

Reply:
It will not get call twice for every index, because as soon we find a null, we exit.

Reply:
@chunhtai I did a mistake when trying to merge/rebase on the latest master version :-(
I just close this PR and open another one https://github.com/flutter/flutter/pull/110177

Reply:
Applying your comments.  And try to be closer to the material specification (by showing the current page when going to the next one....I didn't figure it out how to display the progress indicator without changing the page...I will need help or guidance on this one.